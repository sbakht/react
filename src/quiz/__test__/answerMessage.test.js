import React from 'react';
import { shallow, mount, render } from 'enzyme';
import AnswerMessage from '../answerMessage';
import toJSON from 'enzyme-to-json';

test('render correct', () => {
  var answer = {isCorrect : true} 
  var component = shallow(<AnswerMessage answer={answer} />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('render incorrect', () => {
  var answer = {isWrong : true} 
  var component = shallow(<AnswerMessage answer={answer} />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('has proptypes', () => {
  expect(AnswerMessage.propTypes).toBeDefined();
  
  Object.keys(AnswerMessage.propTypes).map((key) => {
    expect(AnswerMessage.propTypes[key]).toBeDefined();
  });
});