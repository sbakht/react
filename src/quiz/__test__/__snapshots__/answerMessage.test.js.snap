// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correct 1`] = `
<div
  className="answerMsg"
>
  <div
    className="success"
    style={
      Object {
        "color": "green",
        "correct": Object {
          "color": "green",
        },
        "fontWeight": "bold",
        "wrong": Object {
          "color": "red",
        },
      }
    }
  >
    Correct
  </div>
</div>
`;

exports[`render incorrect 1`] = `
<div
  className="answerMsg"
>
  <div
    className="warning"
    style={
      Object {
        "color": "red",
        "correct": Object {
          "color": "green",
        },
        "fontWeight": "bold",
        "wrong": Object {
          "color": "red",
        },
      }
    }
  >
    Incorrect
  </div>
</div>
`;
